geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
+configurations, like allowing customization of the cookie name, could prevent certain implementations from conforming to
+the spec. In other words, opting for a higher-level API provides better interoperability among our implementations.
+
+#### Target Persona
+
+Referring to the [Gateway API Security Model](https://gateway-api.sigs.k8s.io/concepts/security-model/#roles-and-personas),
+the target kubernetes role/persona for session persistence are application developers, as mentioned in the [When does an application require session persistence?](#when-does-an-application-require-session-persistence)
+section. It is the responsibility of the application developers to adjust the persistence configuration to ensure the
+functionality of their applications.
+
+#### Prior Art
+
+Referring to our [Implementations](#Implementations) table on session persistence, the majority of Gateway API
+implementations designed session persistence in their APIs to be attached to a service or backends. This should be
+considered cautiously, as making associations to Gateway API's notion of Gateway, Route, and Service to other
+implementation's objects is hard to directly translate. The idea of a route in Gateway API is often not the same as a
+route in any given implementation.
+
+#### Metaresource Policy Design
+
+In order to apply session persistence configuration to both a service and a route, we will implement it as a
+metaresource policy, as outlined in [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment).
+The metaresource is named `SessionPersistencePolicy` and is only responsible for configuring session persistence for
+services or routes. It is defined as a [Direct Policy Attachment](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#direct-policy-attachment)
+without defaults or overrides, applied to the targeted service, HTTPRoute, or GRPCRoute.
+
+Attaching the `SessionPersistencePolicy` metaresource to a service will be a core support level feature, while attaching
+it to a route will be considered extended. Implementations must support services for conformance, while routes are
+considered optional. This distinction arises because most existing implementations primarily support attaching to a
+service, while attaching to a route is less common and involves greater complexities (see [API Attachment Point: Route vs. Service](#api-attachment-point-route-vs-service)
+for more details).
+
+#### API Attachment Point: Route vs. Service
+
+The `SessionPersistencePolicy` metaresource can target both a service and route object. The core functionality of
+attaching it to either is generally identical. The distinction lies in where the configuration gets propagated: when
+attached to a route, it can define session persistence for multiple services, whereas attaching it to a service defines
+it for a single service.
+
```

Comment:
We need to consider multiple service versions as well as multiple paths under a single service that might have different persistency requirements.

Reply:
Added a paragraph on this. Let me know if I interpreted your statement correctly. It seems like you are saying "all backends **"should"** have session persistence enabled, then say, but implementations need to support this scenario regardless.

> We need to consider multiple service versions as well as multiple paths under a single service that might have different persistency requirements.

For multiple paths under a single service, wouldn't you just "itemize" the route into multiple routes and apply session persistence configuration to the routes that need it? Or this a reference to the suggestion that we should support attaching to the route matching segment?

Reply:
> Or this a reference to the suggestion that we should support attaching to the route matching segment?

Yes, this would make the intentions more explicit

Reply:
Added a reference to targeting route `matcher` section in the future.

Reply:
I think a better way to say this is that you can attach to the route _rule_, not the `matches` section - attaching to the `matches` section wouldn't give you the backends. The rule is the thing that contains `matches`.

Reply:
Good point, thanks @youngnick. I fixed this in the latest commit https://github.com/kubernetes-sigs/gateway-api/pull/2159/commits/3ecf21eba9d9439508f30bb157340d7c4b5b28d9